libsolidity/interface/CompilerStack.cpp

```diff
@@ -260,8 +260,19 @@ bool CompilerStack::analyze()
 	{
 		SyntaxChecker syntaxChecker(m_errorReporter, m_optimiserSettings.runYulOptimiser);
 		for (Source const* source: m_sourceOrder)
-			if (!syntaxChecker.checkSyntax(*source->ast))
+		{
+			if (source->ast)
+			{
+				if (!syntaxChecker.checkSyntax(*source->ast))
+				{
+					noErrors = false;
+				}
+			} else
+			{
 				noErrors = false;
+				return true;
```

Comment:
I would suggest to return `false`. Returning `true`leads to the following output: `Compiler error: Called compile with errors.`. I think we should not call compile an invalid AST anyways, and returning `false` outputs the syntax errors correctly.

Reply:
@erak Sure. This and the other changes when I have to time to get around to it. 

Although I have a test ready, I would also like to test from the javascript bindings. Are there instructions sonewhere on how to build with emscripten?